server/app/views/ViewUtils.java

```diff
@@ -232,4 +234,83 @@ public PTag renderEditOnText(String prefix, Optional<Instant> time) {
+   * @param idPrefix Optional text to use to set IDs for each component. These will be <id>-toggle
+   *                 for the top level button element, <id>-toggle-input for the hidden input field, 
+   *                 <id>-toggle-background for the background of the toggle, and <id>-toggle-nub for
+   *                 the nub inside the toggle.
+   * @param text Optional text label to include with the toggle.
+   * @return ButtonTag containing the toggle.
+   */
+  public static ButtonTag makeToggleButton(
+      String fieldName,
+      boolean enabled,
+      Optional<String> idPrefix,
+      Optional<String> text) {
+    String buttonId = idPrefix.map((v) -> v + "-toggle").orElse("");
+    String inputId = idPrefix.map((v) -> v + "-toggle-input").orElse("");
+    String backgroundId = idPrefix.map((v) -> v + "-toggle-background").orElse("");
+    String nubId = idPrefix.map((v) -> v + "-toggle-nub").orElse("");
+
+    boolean idPresent = idPrefix.isPresent();
+    ButtonTag button =
+        TagCreator.button()
+            .withClasses(
+                "cf-toggle",
+                "flex",
+                "px-0",
+                "gap-2",
+                "items-center",
+                "text-black",
+                "font-normal",
+                "bg-transparent",
+                "rounded-full",
+                StyleUtils.hover("bg-transparent"))
+            .withType("button")
+            .withCondId(idPresent, buttonId)
+            .with(
+                input()
+                    .isHidden()
+                    .withCondId(idPresent, inputId)
+                    .withName(fieldName)
+                    .withClass("cf-toggle-hidden-input")
+                    .withValue(Boolean.valueOf(enabled).toString()))
+            .with(
+                div()
+                    .withClasses("relative")
+                    .with(
+                        div()
+                            .withClasses(
+                                enabled ? "bg-blue-600" : "bg-gray-600",
+                                "w-14",
+                                "h-8",
+                                "rounded-full",
+                                "toggle",
+                                "cf-toggle-background"))
+                            .withCondId(idPresent, backgroundId)
+                    .with(
+                        div()
+                            .withClasses(
+                                "absolute",
+                                "bg-white",
+                                enabled ? "right-1" : "left-1",
+                                "top-1",
+                                "w-6",
+                                "h-6",
+                                "rounded-full",
+                                "cf-toggle-nub")
+                            .withCondId(idPresent, nubId)));
+    if (text.isPresent()) {
+      button.withText(text.get());
+    }
```

Comment:
suggestion (non-blocking): 
```suggestion
    text.ifPresent(button::withText);
```